File: laboratories/ppo/exercises/lab09/php/src/Exceptions/WriteCalledException.php
<?php

namespace Ppo\Lab09\Exceptions;

class WriteCalledException extends InputException
{
    protected $message = ":w character has been provided. Application ends with: :x";
    protected bool $shouldBreak = true;

    public function __construct(string $input)
    {
        $message = str_replace(":x", $input, $this->message);
        parent::__construct($message);
    }
}

File: laboratories/ppo/exercises/lab09/java/lab09/Exceptions/WriteCalledException.java
package lab09.Exceptions;

public class WriteCalledException extends InputException {
  public WriteCalledException(String input) {
    super(":w character has been provided. Application ends with: :x".replace(":x", input));
  }

	@Override
	public boolean shouldBreak() {
		return true;
	}
}
